src/main/java/com/yelp/nrtsearch/server/config/LuceneServerConfiguration.java

```diff
@@ -46,6 +46,7 @@
   private static final Path DEFAULT_PLUGIN_SEARCH_PATH =
       Paths.get(DEFAULT_USER_DIR.toString(), "plugins");
   private static final String DEFAULT_SERVICE_NAME = "nrtsearch-generic";
+  private static final int DEFAULT_ADD_DOCUMENTS_MAX_BUFFER_LEN = 100;
```

Comment:
The PR seems fine but I was wondering if we want to make the MAX_BUFFER_LEN as per jvm start config like you do here or should we make it a part of the AddDocumentRequest so that each indexing request  could have a different setting and we do not need to restart the server for this to take effect. One example would be to run a backfill vs real time indexing.